Hopefully fix deployment

DRMacIver opened the pull request:
See #589 - deploys currently think they're not on master when they are.

This updates the master check to use "git merge-base --is-ancestor", which is hopefully more correct.

It's obviously a little hard to test this without actually deploying, but this should work.

alexwlchan commented on scripts/deploy.py:
```diff
@@ -76,7 +76,12 @@
         sys.executable, 'setup.py', 'sdist', '--dist-dir', DIST,
     ])
 
-    if not tools.on_master():
```
This was the only use of `on_master()`. Worth deleting it as unused?

DRMacIver replied:
Yes

alexwlchan commented on scripts/hypothesistooling.py:
```diff
@@ -82,6 +82,12 @@ def hash_for_name(name):
 
+def is_ancestor(a, b):
+    return subprocess.call([
+        'git', '--is-ancestor', a, b
```
> This updates the master check to use "git merge-base --is-ancestor"

🤔

DRMacIver replied:
I don't really understand why the old logic doesn't work either. But it's likely I've just misunderstood git's merge-base semantics.

alexwlchan replied:
I should have been more explicit about the problem – the command you’ve implemented doesn’t include the `merge-base` verb.

DRMacIver replied:
\*headdesk\*

Right. Lets fix that...

DRMacIver commented:
@alexwlchan All changes done I think.

DRMacIver commented:
@alexwlchan OK I've fixed the missing verb and also added a check so that if I've got the git command wrong then it will error sensibly rather than silently doing the wrong thing (the documentation explicitly says that errors are signalled by any status code not in (0, 1), so that assertion is correct)
